litho-core/src/main/java/com/facebook/litho/DefaultInternalNode.java

```diff
@@ -158,6 +158,7 @@
   private @Nullable ArrayList<Component> mComponentsNeedingPreviousRenderData;
   private @Nullable ArrayList<WorkingRangeContainer.Registration> mWorkingRangeRegistrations;
   private @Nullable String mTestKey;
+  private @Nullable Object mComponentTag;
```

Comment:
I would recommend changing this into an `Map<Component, Object>` instead. The reasoning is similar to why `mComponents` is a list; because Litho flattens the hierarchy, an InternalNode can represent more than one Component.

Reply:
Actually, I don't think we need to copy the component tag into the InternalNode, since access via the DebugComponent is only done through the CommonProps, which also stores the tag. I'll most likely remove this from InternalNode and related classes. If there is some need for accessing the component tag in an Internal Node, it can be added in the future.

Reply:
I agree, if for the use case you are exploring, you can can access this via `Component.getCommonProps().getComponentTag()` then let us avoid add this to the `InternalNode`.

Reply:
Removed component tag from `InternalNode`, should hopefully be a lot simpler now. Thanks for the review btw!